### PR TITLE
test: goals tests fixes

### PR DESCRIPTION
**Pull request by zwsaqx**
- [Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/36562260/todos/7145954621)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Goals.spec.tsx passes
- [ ] Goalslist.spec.tsx passes
- [ ] GoalDetails.spec.tsx passes

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Slider/Content/Goals/GoalsList/GoalsList.spec.tsx [hunk 1]
```diff
@@ -68,65 +67,36 @@ describe('GoalsList', () => {
...
-      )
+    it('should call dispatch with a URL when "Learn More" button is clicked', () => {
+      const { getByRole } = render(<GoalsList goals={goals} />)
       fireEvent.click(getByRole('button', { name: 'Learn More' }))
-      expect(getByText('Information')).toBeInTheDocument()
-      fireEvent.click(getByTestId('X2Icon'))
-      expect(getByTestId('GoalInformation').parentElement).not.toHaveClass(
-        'MuiDrawer-parentAnchorRight'
-      )
+      expect(useEditor().dispatch).toHaveBeenCalledWith({
```
Comment (jianwei1) [inline apps/journeys-admin/src/components/Editor/Slider/Content/Goals/GoalsList/GoalsList.spec.tsx:79]:
```suggestion
      expect(dispatch).toHaveBeenCalledWith({
```
Can we change this just to 'dispatch', in order to keep consistent with line 93?

File: apps/journeys-admin/src/components/Editor/Slider/Content/Goals/GoalsList/GoalsList.spec.tsx [hunk 2]
```diff
@@ -19,7 +18,7 @@ import { GoalsList } from './GoalsList'
 
 jest.mock('@mui/material/useMediaQuery', () => ({
   __esModule: true,
-  default: jest.fn()
+  default: jest.fn(() => false)
```
Comment (jianwei1) [inline apps/journeys-admin/src/components/Editor/Slider/Content/Goals/GoalsList/GoalsList.spec.tsx:21]:
```suggestion
  default: jest.fn()
```
Do we need to return false here? the tests pass whether its empty or false

File: apps/journeys-admin/src/components/Editor/Slider/Content/Goals/Goals.spec.tsx [hunk 3]
```diff
@@ -167,19 +166,19 @@ describe('GoalsList', () => {
           variant: 'admin'
         }}
       >
-        <GoalsList goals={[]} />
+        <Goals />
       </JourneyProvider>
     )
     expect(getByText('The Journey Goals')).toBeInTheDocument()
 
     expect(getAllByText('https://m.me/some_user')[0]).toBeInTheDocument()
```
Comment (jianwei1) [inline apps/journeys-admin/src/components/Editor/Slider/Content/Goals/Goals.spec.tsx:174]:
```suggestion
    expect(getByText('https://m.me/some_user')).toBeInTheDocument()
```
Can these also be changed to getByText?

**Review by jianwei1 — CHANGES_REQUESTED** [review]

**Review by jianwei1 — APPROVED** [review]
